### ISSUE TITLE
Feature Request: Server Side Rendering

### ISSUE THREAD

**Issue by prashanth-hegde** [khalyomede/mantis#3]
Hello, 

I have looked at mantis, and it looks very promising. 
There is a looong list of features that are planned, and I am excited to try them. 

Here is a feature request, that I think adds value to mantis. 

1. Server Side Rendering
  1.1. Templating
  1.2. Support for htmx (possibly)

**Comment by khalyomede** [comment]
Hello,

I'm very glad this project appeals to you. It's early days for Mantis and
I'm super pumped to work on it as I have received many positive feedbacks
including yours!

I just noted HTMX support to be developed in priority.

For templating, do you find yourself more productive with separate .html or
do you like the full .v file approach? This will help me gather needs to
improve the roadmap.

Thank you very much for the time you took reaching out.
Please make sure to follow me on Twitter for updates on the next features:
https://x.com/anwar_nairi


See you around 👋

Le jeu. 26 déc. 2024, 21:31, encyclopaedia ***@***.***> a
écrit :

**Comment by prashanth-hegde** [comment]
The current veb implementation, although has a lot of features, is confusing to setup. And is not intuitive.

I, personally like a separate .html approach because I think provides a clean separation. We may just need a simple static file server + support for `if` and `for` conditionals in the templates

**Comment by khalyomede** [comment]
That's an interesting thing to tackle, I will see if I can come up with an idea that do not give up on type safety. I will keep you posted on 1.1 soon.

1.2 is planned, thank you for raising the priority on it :pray:

**Comment by aminnairi** [comment]
I don't think that supporting HTMX should be in the scope of this project, because if Mantis supports HTMX, why not Alpine? Or Inertia? Or rather why? Mantis should stick to what is has been designed: create an HTTP Web server and if we want to add Alpine, Inertia, HTMX or [Insert the newest trendy framework here], it should be user land, not on the framework core scope.

Having a templating syntax like it is done in Blade (Laravel) or Twig (Symfony) is not something that aligns with the philosophy of the language nor the framework in my opinion.

PHP, Python, JavaScript are well suited for having templating engine as they are interpreted languages with no compilation steps nor compile-time checks. I think that having a templating syntax baked-in the language like this is done currently (not in the flavor of a templating engine that relies on static files) is better since it provide actual and useful compile-time checks like forgetting to provide a value needed for the template, or mistakes by providing an incorrect template file path since there is nothing like that, it is just pure functions, with the language type checking and no nonsense HTML file with sprinkles of weird syntax that turns into a string that has no compile-time checking opening a wide spectrum of human errors with no fool-proofing.

Just my 2 cents on the question.

**Comment by khalyomede** [comment]
I actually agree on this take: if we want to follow the philosophy of the framework, third-party front-end tooling is out-of-scope of **this repository**.

In the mean time I am brainstorming on making it even easier to get started with Mantis.

Code name: Mantis Start.

This would be an opiniated set of repositories with flavored defaults. One of them could be featuring HTMX.

So for example you could have:

- Mantis Start - basic (Tailwind/Mantis HTML)
- Mantis Start - Inertia React
- Mantis Start - Inertia Vue
- Mantis Start - HTMX
- ...

So ultimately this issue will be closed and reopened once Mantis Start is ready. For the moment I will focus effort on solidifying the core, and once this repository (library) roadmap is well advanced I will introduce Mantis Start.

The reason it is not done yet is because once you create these repositories, they need to follow the packages updates (this means following news on HTMX, Inertia, Tailwind, ...) which is an additional time I cannot buy for now.

I probably see Mantis Start being championed by people willing to maintain the flavored repositories in the future, but this should be discussed in a future RFC in my opinion.

Anyway good input @aminnairi thanks!
